torchsynth/module.py

```diff
@@ -405,8 +423,9 @@ def _forward(self, mod_signal: Signal) -> Signal:
 
         assert (mod_signal >= -1).all() and (mod_signal <= 1).all()
         control_as_frequency = self.make_control_as_frequency(mod_signal)
-        cosine_argument = self.make_argument(control_as_frequency) + self.phase
-        self.phase = cosine_argument[:, -1]
+        cosine_argument = self.make_argument(control_as_frequency)
+        cosine_argument += self.phase.unsqueeze(1)
```

Comment:
Later but not now we should figure out if unsqueeze is a view (fast) or clone (slow)

Reply:
it's a view: "The returned tensor shares the same underlying data with this tensor."